Smaller aesthetic fixes to InternalTestCluster

ywelsch opened the pull request:
Smaller stuff that I noticed while working on other stuff...

ywelsch committed:
Smaller aesthetic fixes to InternalTestCluster

elasticmachine commented:
Pinging @elastic/es-core-infra

DaveCTurner commented on test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java:
```diff
@@ -1649,10 +1638,7 @@ public synchronized String startNode(Settings.Builder settings) {
      * Starts a node with the given settings and returns it's name.
```
*its (it's wrong in some other places too)

DaveCTurner reviewed (commented):
Changes LGTM, but tests are failing. Not sure if related or not.

ywelsch committed:
Handle rapid expansion of master-eligible nodes

Adding too many master-eligible nodes at once can fail in setting the min_master_ndoes property before the nodes join

ywelsch committed:
it's

ywelsch commented:
The test failures were actually caused by this PR. Due to the now possibly more rapid expansion of clusters, we cannot always update the min_master_node setting before starting the new master-eligible nodes.

Assume for example that you have a one node cluster and you add two more nodes. We can then not set minimum_master_nodes to 2 on that single-node cluster, as we have a settings validation check that ensures that minimum_master_nodes cannot be set to a value higher than the number of master-eligible nodes in the cluster.

DaveCTurner reviewed (approved):
Makes sense, LGTM.
